backend/pkg/handler/groupMemberHandler.go

```diff
@@ -5,228 +5,514 @@ import (
+		return
+	}
+
+	// This deletes the invitation from the database
+	err = deleteGroupInvitation(h.invitationRepo, id)
+	if err != nil {
+		// Handle the error if deleting the invitation fails.
+		http.Error(w, "Failed to delete invitation from the database: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+}
+
+// -------- Notification Functions -------- //
+
+// NotifyInvitationDecline notifies the sender about the declined invitation.
+func notifyInvitationDecline(invitationRepo *repository.InvitationRepository, notificationRepo *repository.NotificationRepository, id string) error {
+	// Get the invitation details from the repository based on the ID.
+	invitation, err := invitationRepo.GetGroupInvitationByID(id)
+	if err != nil {
+		return err
+	}
+
+	// Construct a notification message.
+	message := fmt.Sprintf("Your invitation to join the group %d has been declined by the user %d.", invitation.GroupId, invitation.JoinUserId)
+
+	// Create a new notification.
+	newNotification := model.Notification{
+		UserId:    invitation.InviteUserId, // Sender's user ID
+		Type:      "invitation_declined",
+		Message:   message,
+		IsRead:    false,
+		CreatedAt: time.Now(),
+	}
+
+	// Add the notification to the database.
+	_, err = notificationRepo.CreateNotification(newNotification)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ----------------------------------------------------------------------------------------------------------
+
+// GetGroupInvitationByIDHandler gets an invitation by ID for the user.
 func (h *GroupMemberHandler) GetGroupInvitationByIDHandler(w http.ResponseWriter, r *http.Request) {
-    vars := mux.Vars(r)
-    id := vars["id"]
-    invitation, err := h.invitationRepo.GetGroupInvitationByID(id)
-    if err != nil {
-        http.Error(w, "Failed to get invitation: "+err.Error(), http.StatusInternalServerError)
-        return
-    }
-    json.NewEncoder(w).Encode(invitation)
+	// Extract the user ID from the cookie.
+	userID, err := h.sessionRepo.GetUserIDFromSessionToken(util.GetSessionToken(r))
+	if err != nil {
+		http.Error(w, "Error extracting user ID from session token: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	vars := mux.Vars(r)
+	id := vars["id"]
+
+	// Check if the user has the permission to view the invitation.
+	invitation, err := h.invitationRepo.GetGroupInvitationByID(id)
+	if err != nil {
+		http.Error(w, "Failed to get invitation: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if invitation.JoinUserId != userID {
+		http.Error(w, "User does not have permission to view this invitation", http.StatusUnauthorized)
+		return
+	}
+
+	// Encode the invitation in the response.
+	json.NewEncoder(w).Encode(invitation)
 }
 
-// GetAllInvitationsHandler gets all invitations (ok)
+// GetAllGroupInvitationsHandler gets all pending invitations for the user.
```

Comment:
there should also be a function for the group owner to see all of the groups invitations and requests (doesn't matter which status)

Reply:
ok maybe invitations and requests with accepted status give no value so only pending and declined